src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs

```diff
@@ -100,7 +100,31 @@ public async Task InitializedAsync()
         }
 
         [JsonRpcMethod(Methods.ShutdownName)]
-        public object? Shutdown(CancellationToken _) => null;
+        public Task ShutdownAsync(CancellationToken _)
+        {
+            _diagnosticService.DiagnosticsUpdated -= DiagnosticService_DiagnosticsUpdated;
+
+            return Task.CompletedTask;
+        }
+
+        [JsonRpcMethod(Methods.ExitName)]
+        public Task ExitAsync(CancellationToken _)
+        {
+            try
```

Comment:
Can we ensure that shutdown has been called first?  If there's a bug and the client doesn't call shutdown before exit, I want to make sure it's easy to spot rather than just continuing with success (and not unsubscribing)